vault/identity_store_oidc.go

```diff
@@ -343,20 +395,30 @@ func (i *IdentityStore) getOIDCConfig(ctx context.Context, s logical.Storage) (*
 
 	c.effectiveIssuer = c.Issuer
 	if c.effectiveIssuer == "" {
-		c.effectiveIssuer = i.core.redirectAddr + issuerPath
+		c.effectiveIssuer = i.core.redirectAddr
 	}
 
-	i.oidcCache.SetDefault("config", &c)
+	c.effectiveIssuer += "/" + ns.Path + issuerPath
```

Comment:
c.effectiveIssuer = path.Join(c.effectiveIssuer, ns.Path, issuerPath)

Reply:
Yep, that'll work. I thought that it would result in `//` if `ns.Path`was "", but it looks like it handles that case the way we'd like.

Reply:
Actually, it won't work 😕. The slashes in "https://" get mangled: https://play.golang.org/p/yW-rHnaxEBt

Reply:
Hah, I didn't realize that the first bit is an URL.